packages/babel-helper-function-name/src/index.ts

```diff
@@ -197,9 +197,28 @@ function visit(
 }
 
 /**
- * @param {NodePath} param0
- * @param {Boolean} localBinding whether a name could shadow a self-reference (e.g. converting arrow function)
- * @param {Boolean} supportUnicodeId whether a target support unicodeId or not
+ * Add id to function/class expression inferred from the AST
+ *
+ * @export
+ * @template N The unamed expression type
+ * @param {({
+ *     node: N;
+ *     parent?: NodePath<N>["parent"];
+ *     scope: Scope;
+ *     id?: t.LVal | t.StringLiteral | t.NumericLiteral | t.BigIntLiteral;
+ *   })} {
+ *     node,
+ *     parent,
+ *     scope,
+ *     id,
+ *   } `node`, `parent` and `scope` are generally extracted from a given NodePath. `id` is the fallback naming source when the helper
+ * can not infer the function name from the AST
```

Comment:
JSDoc does not support destructuring, but we can do something like this:
```suggestion
 * @param {NodePath<t.FunctionExpression | t.Class>} The function or class to name.
```

Also, `id` is not a fallback: if there is already an id we exit early, and the fallback behavior is to _infer_ it.

Reply:
If we have `node.id`, then `node` is a named function/class expression, we exit early.
Then the helper tries to determine `id` from the AST, and if the fallback `id` is not provided, we exit early.
Then the helper determine a name from `id`, exit early if the name can't be determined.

In this way the `id` is the fallback naming source. And because `id` does not exist in a NodePath, we can't put `NodePath` here.